### ISSUE TITLE
IDE plugin for visualizing the navigation graphs

### ISSUE THREAD

**Issue by MohamedRejeb** [raamcosta/compose-destinations#617]
Hi,
I think that having an IDE plugin for visualizing the navigation graphs is a good idea, what do you think?
I want to know if there is any technical limitation that prevents having it.
I can work on implementing this feature, let me know if you have any advice or things I need to know before I start.

**Comment by raamcosta** [comment]
Hi @MohamedRejeb 

I have thought about this numerous times, but haven't had time to properly investigate and explore it.

I would say, KSP task could provide some output file with details of the graphs, and the plugin would read it and display the interactive nav graph.
KSP task is already outputting stuff like mermaid files for the structure of the graph, so either make use of that file (if the formatting is good enough --  I guess if the plugin used JS, it could just render the mermaid file directly 🤔 ?).

Let me know if I can help 🤞

**Comment by GuilhE** [comment]
We can use this one: https://plugins.jetbrains.com/plugin/20146-mermaid but for some reason is not opening the preview tab 😓

**Comment by raamcosta** [comment]
I don’t think it works on Android studio unfortunately. At least not in with the version I tested on.

**Comment by GuilhE** [comment]
> I don’t think it works on Android studio unfortunately. At least not in with the version I tested on.
![Screenshot 2024-05-20 at 12 32 44](https://github.com/raamcosta/compose-destinations/assets/2677139/7a7c3fa5-9e40-441a-b7a0-ee2ff38cd4fe)

Correct, it works with IntelliJ IDEA CE but not with Android Studio 😢

**Comment by FooIbar** [comment]
> Correct, it works with IntelliJ IDEA CE but not with Android Studio

You need to install JBR with JCEF for it to work on AS
https://issuetracker.google.com/issues/159933628#comment7
